next.config.ts

```diff
@@ -1,7 +1,26 @@
-import type { NextConfig } from 'next';
+import type { NextConfig } from "next";
 
 const nextConfig: NextConfig = {
-  /* config options here */
+  async headers() {
+    return [
+      {
+        source: "/api/:path*",
+        headers: [
+          { key: "Access-Control-Allow-Credentials", value: "true" },
+          { key: "Access-Control-Allow-Origin", value: "*" }, // 개발 중에는 모든 출처(*)를 허용합니다.
```

Comment:
CORS(교차 출처 리소스 공유)의 `Access-Control-Allow-Origin `헤더 설정을 * 로 설정 하게 되면 출처에 상관없이 리소스에 접근할 수 있기에 보안이 취약해 질 것 같습니다. 

그래서  `Access-Control-Allow-Origin : https: exampl .com` 처럼 직접 허용할 출처를 세팅하는 방법이 좋을 것 같습니다.

[토스 payments](https://docs.tosspayments.com/resources/glossary/cors)에서도 출처에 상관없이 리소스에 접근하게 될때에 대해서 이야기 하고 있습니다. 
참고 해주시면 감사합니다. 

참고
> [토스 payments](https://docs.tosspayments.com/resources/glossary/cors)

Reply:
현재 설정 `value: *`은 모든 도메인 요청을 허용하는 이유는 개발 환경에서 편의성을 했습니다. 하지만, 실제 서비스 운영할 때는 **허가된 고객사 도메인**만 요청을 보낼 수 있도록 하는것이 토스 자료처럼 보안상 안전합니다.

그리고 허가된 도메인도 코드상에 직접 보여주기 보다 환경변수 `.env`파일에 허용할 고객사 도메인 목록을 관리하면 더 보안상을 좋을거 같습니다
```suggestion
          { key: "Access-Control-Allow-Origin", value: process.env.DESKER_CLIENTS_DOMAIN },
```

일단 개발단계에서는 환경변수 `DESKER_CLIENTS_DOMAIN`을 값을 `*` 임시로 값으로 하고 배포할 때 정해진 고객사 도메인 값들을 넣어주면 좋을거 같아요

Reply:
넵 좋습니다.. 저도 작업하면서 중요한 데이터들은 환경변수`.env` 파일에 넣어 관리 하겠습니다.